core/src/main/java/org/apache/iceberg/rest/responses/RESTCatalogConfigResponse.java

```diff
@@ -0,0 +1,154 @@
+ * created for use with the catalog.
+ * <p>
+ * Configuration from the server consists of two sets of key/value pairs.
+ * <ul>
+ *   <li> defaults - properties that should be used as default configuration </li>
+ *   <li> overrides - properties that should be used to override client configuration </li>
+ * </ul>
+ */
+public class RESTCatalogConfigResponse {
+
+  private Map<String, String> defaults;
+  private Map<String, String> overrides;
+
+  public RESTCatalogConfigResponse() {
+    // Required for Jackson deserialization
+  }
+
+  private RESTCatalogConfigResponse(Map<String, String> defaults, Map<String, String> overrides) {
+    this.defaults = defaults;
+    this.overrides = overrides;
+    validate();
+  }
+
+  RESTCatalogConfigResponse validate() {
+    return this;
+  }
+
+  /**
+   * Properties that should be used as default configuration. {@code defaults} have the lowest priority
+   * and should be applied before the client provided configuration.
+   *
+   * @return properties that should be used as default configuration
+   */
+  public Map<String, String> defaults() {
+    return defaults != null ? Maps.newHashMap(defaults) : Collections.emptyMap();
+  }
+
+  /**
+   * Properties that should be used to override client configuration. {@code overrides} have the highest priority
+   * and should be applied after defaults and any client-provided configuration properties.
+   *
+   * @return properties that should be given higher precedence than any client provided input
+   */
+  public Map<String, String> overrides() {
+    return overrides != null ? Maps.newHashMap(overrides) : Collections.emptyMap();
+  }
+
+  /**
+   * Merge client-provided config with server side provided configuration to return a single
+   * properties map which will be used for instantiating and configuring the REST catalog.
+   *
+   * @param clientProperties - Client provided configuration
+   * @return Merged configuration, with precedence in the order overrides, then client properties, and then defaults.
+   */
+  public Map<String, String> merge(Map<String, String> clientProperties) {
+    Preconditions.checkNotNull(clientProperties,
+        "Cannot merge client properties with server-provided properties. Invalid client configuration: null");
+    Map<String, String> merged = Maps.newHashMap(defaults);
+    merged.putAll(clientProperties);
+    merged.putAll(overrides);
+    return merged;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("defaults", defaults)
+        .add("overrides", overrides)
+        .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private final Map<String, String> defaults;
+    private final Map<String, String> overrides;
+
+    private Builder() {
+      this.defaults = Maps.newHashMap();
+      this.overrides = Maps.newHashMap();
```

Comment:
You could use immutable builders like the other classes.

Reply:
Using immutable builders makes it so that a `null` value can't be set. Which people might want to do for `overrides` as one example to explicitly disable some configuration .

For example, if an operator wanted to remove all client provided values for a given setting across all jobs for something temporarily, to maybe investigate too many connections to the catalog service or something. They could remove the clients configuration by placing `clients: null` into `overrides` and then setting `clients: 2` in their defaults. Somewhat of a contrived example, but at scale when operators and users are two separate groups, I think this could very much have a practical use case.

Reply:
Oh, very good point. +1 for keeping it `HashMap` here.